core/src/main/java/org/infinispan/protostream/annotations/impl/types/ReflectionClassFactory.java

```diff
@@ -109,9 +111,9 @@ public XClass fromTypeMirror(TypeMirror typeMirror) {
                   try {
                      f.setAccessible(true);
                      e = (Enum<?>) f.get(clazz);
-                  } catch (IllegalAccessException iae) {
+                  } catch (IllegalArgumentException | IllegalAccessException ex) {
```

Comment:
I'd rather not handle IllegalArgumentException or any other exceptions here, just let it explode. But IllegalAccessException is not a runtime exception.

Reply:
Got it, I'll roll it back. Point was to throw an exception with useful information.

Reply:
reverted.